### DISCUSSION TITLE
Type clipboard like a human

### DISCUSSION THREAD

**Discussion by trevor-atlas** [johnlindquist/kit#1295] [Share]
I haven't quite dialed in the random delays as well as I'd like, but it gets the job done :]

[Open humanlike-typing in Script Kit](https://scriptkit.com/api/new?name=humanlike-typing&url=https://gist.githubusercontent.com/trevor-atlas/17746a243dd9bbfa8062d8fb86b5fc20/raw/45b2a5cb769b76db73f70579edf28b469ba194bd/humanlike-typing.ts")

```js
// Name: humanlike typing
// Description: Type the contents of your clipboard as if you were a human
// Author: Trevor Atlas
// Twitter: @trevoratlas

import "@johnlindquist/kit"

await hide();

await applescript(String.raw`
set texttowrite to the clipboard as text
tell application "System Events"
  repeat with i from 1 to count characters of texttowrite
    if (character i of texttowrite) is equal to linefeed or (character i of texttowrite) is equal to return & linefeed or (character i of texttowrite) is equal to return then
      keystroke return
    else
      keystroke (character i of texttowrite)
    end
    if (character i of texttowrite) is equal to " " then
      delay (random number from 0.01 to 0.1)
    else if (character i of texttowrite) is equal to "\n" then
      delay (random number from 0.1 to 0.3)
    else
      delay (random number from 0.01 to 0.05)
    end
  end repeat
end tell
`);

```

**Comment by Strajk** [comment]
@trevor-atlas genuinely curious... in what situations do u use this? 🥷

**Reply by trevor-atlas** [reply]
mostly for presenting code, it makes slides a little more interesting than just staring at a static block of text

**Reply by ramiroaraujo** [reply]
Fun fact: I have a similar script (former Alfred workflow), and I use it mostly for inputs where the paste shortcut doesn't work correctly. There're not a ton, but there are. In my case I use a very short delay, it's _almost_ a paste :)
